.github/workflows/ci.yml

```
@@ -45,54 +34,119 @@ jobs:
       - name: Build all packages
         run: npm run build
 
+  get-commands:
+    name: Get scaffold commands
+    if: "!contains(github.event.head_commit.message, 'ci skip') && !contains(github.event.head_commit.message, 'skip ci')"
+    runs-on: ubuntu-latest
+    outputs:
+      scaffoldCommands: ${{ steps.set-matrix.outputs.scaffoldCommands }}
+    steps:
+      - id: set-matrix
+        run: echo "scaffoldCommands=[{\"cmd\":\"generate-app\",\"hasBackend\":false},{\"cmd\":\"generate-app-backend\",\"hasBackend\":true},{\"cmd\":\"generate-panel\",\"hasBackend\":false},{\"cmd\":\"generate-datasource\",\"hasBackend\":false},{\"cmd\":\"generate-datasource-backend\",\"hasBackend\":true},{\"cmd\":\"generate-scenes-app\",\"hasBackend\":false}]" >> $GITHUB_OUTPUT
+
+  generate-plugins:
+    name: Test plugin scaffolding
+    if: "!contains(github.event.head_commit.message, 'ci skip') && !contains(github.event.head_commit.message, 'skip ci')"
+    needs: [test, get-commands]
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        scaffoldCommand: ${{ fromJson(needs.get-commands.outputs.scaffoldCommands) }}
+    env:
+      NX_CLOUD_ACCESS_TOKEN: ${{ secrets.NX_CLOUD_ACCESS_TOKEN }}
+      NX_BRANCH: ${{ github.event.number || github.ref_name }}
+    steps:
+      - run: echo ${{ matrix.scaffoldCommand.cmd }}
+      - run: echo ${{ matrix.scaffoldCommand.hasBackend }}
```

Nit:
```suggestion
      - run: echo "Command: ${{ matrix.scaffoldCommand.cmd }}"
      - run: echo "Has backend: ${{ matrix.scaffoldCommand.hasBackend }}"
```
But you can also remove this, seems more for debugging purposes